pickhardtpayments/OracleLightningNetwork.py

```diff
@@ -78,22 +121,41 @@ def theoretical_maximum_payable_amount(self, source: str, destination: str, base
         mincut, _ = nx.minimum_cut(test_network, source, destination)
         return mincut
 
-    def settle_payment(self, path: List[OracleChannel], payment_amount: int):
+    def settle_attempt(self, attempt: Attempt):
         """
-        receives a List of channels and payment amount and adjusts the balances of the channels along the path.
+        receives a payment attempt and adjusts the balances of the OracleChannels and its reverse channels
+        along the path.
 
-        settle_payment should only be called after all send_onions for a payment terminated successfully!
-        # TODO testing
+        settle_attempt should only be called after all send_onions for a payment terminated successfully!
         """
-        for channel in path:
+        for channel in attempt.path:
             settlement_channel = self.get_channel(channel.src, channel.dest, channel.short_channel_id)
             return_settlement_channel = self.get_channel(channel.dest, channel.src, channel.short_channel_id)
-            if settlement_channel.actual_liquidity > payment_amount:
+            if settlement_channel.actual_liquidity > attempt.amount:
                 # decrease channel balance in sending channel by amount
-                settlement_channel.actual_liquidity = settlement_channel.actual_liquidity - payment_amount
+                settlement_channel.actual_liquidity = settlement_channel.actual_liquidity - attempt.amount
+                # remove in_flight amount
+                settlement_channel.in_flight -= attempt.amount
                 # increase channel balance in the other direction by amount
-                return_settlement_channel.actual_liquidity = return_settlement_channel.actual_liquidity + payment_amount
+                if return_settlement_channel:
+                    return_settlement_channel.actual_liquidity = return_settlement_channel.actual_liquidity \
+                                                                 + attempt.amount
             else:
                 raise Exception("""Channel liquidity on Channel {} is lower than payment amount.
                     \nPayment cannot settle.""".format(channel.short_channel_id))
         return 0
+
+    def remove_channels_with_no_return_channels(self):
+        ebunch = []
```

Comment:
I do not understand the variable name

Reply:
the variable name directly stems from the use in the networkx library examples. I stuck with it for consistency reasons in the context of the networkx system, but of course can be changed to something more meaningful.